From 3107d07898136ac85058ef9b0d45b490f85f9205 Mon Sep 17 00:00:00 2001
From: Josh Poimboeuf <jpoimboe@redhat.com>
Date: Thu, 4 Sep 2014 18:54:41 -0400
Subject: [s390] ftrace: remove check of obsolete variable function_trace_stop

Message-id: <38a88f36925e14f76f2248ef2fcfdc12213a8134.1409856554.git.jpoimboe@redhat.com>
Patchwork-id: 91258
O-Subject: [RHEL7.1 PATCH 22/23] s390/ftrace: remove check of obsolete variable function_trace_stop
Bugzilla: 1113829
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1113829

commit 2a62a57a0633c7bbc9689d738ec0235e2ec822dc
Author: Heiko Carstens <heiko.carstens@de.ibm.com>
Date:   Thu Jun 26 10:43:25 2014 +0200

    s390/ftrace: remove check of obsolete variable function_trace_stop

    Remove check of obsolete variable function_trace_stop as requested by
    Steven Rostedt.

    Signed-off-by: Heiko Carstens <heiko.carstens@de.ibm.com>
    Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/s390/Kconfig b/arch/s390/Kconfig
index 0669656..f800b80 100644
--- a/arch/s390/Kconfig
+++ b/arch/s390/Kconfig
@@ -114,7 +114,6 @@ config S390
  select HAVE_FTRACE_MCOUNT_RECORD
  select HAVE_FUNCTION_GRAPH_TRACER
  select HAVE_FUNCTION_TRACER
- select HAVE_FUNCTION_TRACE_MCOUNT_TEST
  select HAVE_KERNEL_BZIP2
  select HAVE_KERNEL_GZIP
  select HAVE_KERNEL_LZMA
diff --git a/arch/s390/kernel/mcount.S b/arch/s390/kernel/mcount.S
index 08dcf21..433c6db 100644
--- a/arch/s390/kernel/mcount.S
+++ b/arch/s390/kernel/mcount.S
@@ -21,13 +21,9 @@ ENTRY(_mcount)
 ENTRY(ftrace_caller)
 #endif
  stm %r2,%r5,16(%r15)
- bras %r1,2f
+ bras %r1,1f
 0: .long ftrace_trace_function
-1: .long function_trace_stop
-2: l %r2,1b-0b(%r1)
- icm %r2,0xf,0(%r2)
- jnz 3f
- st %r14,56(%r15)
+1: st %r14,56(%r15)
  lr %r0,%r15
  ahi %r15,-96
  l %r3,100(%r15)
@@ -50,7 +46,7 @@ ENTRY(ftrace_graph_caller)
 #endif
  ahi %r15,96
  l %r14,56(%r15)
-3: lm %r2,%r5,16(%r15)
+ lm %r2,%r5,16(%r15)
  br %r14
 
 #ifdef CONFIG_FUNCTION_GRAPH_TRACER
diff --git a/arch/s390/kernel/mcount64.S b/arch/s390/kernel/mcount64.S
index 1c52eae..c67a8bf 100644
--- a/arch/s390/kernel/mcount64.S
+++ b/arch/s390/kernel/mcount64.S
@@ -20,9 +20,6 @@ ENTRY(_mcount)
 
 ENTRY(ftrace_caller)
 #endif
- larl %r1,function_trace_stop
- icm %r1,0xf,0(%r1)
- bnzr %r14
  stmg %r2,%r5,32(%r15)
  stg %r14,112(%r15)
  lgr %r1,%r15
-- 
1.7.1